### PR TITLE
Replace scrollHeight with children height in ExpandableTransition.expand

### PR DESCRIPTION
**Pull request by Scotchester**
The purpose of this is to enable expandable heights to be recalculated while open to be _shorter_. Using the previous `scrollHeight` method, open expandables can call `expand` to _grow_ if their contents have changed, but if their contents have shrunk, `scrollHeight` still returns the rendered height from when it was at its maximum.

This is necessary for the correct height adjustments as the multiselect filters are used within a Filterable List Control expandable in cfgov-refresh (e.g., on the [Blog](https://www.consumerfinance.gov/about-us/blog/)).

## Changes

- **cf-expandables:** [PATCH] Replace `scrollHeight` with children height in `ExpandableTransition.expand`

## Testing

- To test that this doesn't negatively impact basic expandables, follow the [local testing instructions](https://github.com/cfpb/capital-framework/blob/canary/CONTRIBUTING.md#testing-in-the-documentation-site) and look at the Expandables page in the docs site.
- `gulp test`

## Todos

- Open the PR in cfgov-refresh to test this on Filterable List Controls.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Internet Explorer 8, 9, 10, and 11
- [x] Edge
- [x] iOS Safari
- [x] Chrome for Android

### CODE REVIEW

**Comment by Scotchester** [comment]
Question for the group:

Tests still pass, but I can't tell if I need to add a test for the change in functionality or not.

File: src/cf-expandables/src/ExpandableTransition.js [hunk 1]
```diff
@@ -98,8 +98,14 @@ function ExpandableTransition( element ) {
   function expand() {
     this.dispatchEvent( 'expandBegin', { target: this } );
 
-    if ( !previousHeight || element.scrollHeight > previousHeight ) {
-      previousHeight = element.scrollHeight;
+    let childrenHeight = 0;
+    for ( const child of element.children ) {
```
Comment (anselmbradford) [inline src/cf-expandables/src/ExpandableTransition.js]: No IE support for `for … of`. Do we care? It may also be picked up by babel.

**Comment by anselmbradford** [comment]
> Tests still pass, but I can't tell if I need to add a test for the change in functionality or not.

For animated features that coordinate with CSS we need acceptance tests, which we don't have in CF.
We could move the code you added to a separate public method (`recalculateHeight()` or something), and test that it returns expected heights with different html snippets of various heights. However, that would appear to be making a public method only for the sake of the tests, which doesn't seem right. I'd only go that way if there's a use-case for a `recalculateHeight()` method outside of the tests, otherwise acceptance tests would be the way to go.

File: src/cf-expandables/src/ExpandableTransition.js (hunk 1 above)
Comment (Scotchester) [reply src/cf-expandables/src/ExpandableTransition.js]: Ah, right. I'll change it to `Array.prototype.forEach.call`.

**Comment by Scotchester** [comment]
Browser/acceptance tests would be good, but as we have none currently set up for Capital Framework, I'll defer that to later.

**Pull request merged**